api/types.proto

```diff
@@ -793,12 +793,10 @@ message SecretReference {
 	string secret_name = 4;
 }
 
-// RemovedNode is a record for a node that has been removed from the swarm.
-message RemovedNode {
-	// ID is the ID of the removed node.
-	string id = 1 [(gogoproto.customname) = "ID"];
-
+// BlacklistedCertificate is a record for a blacklisted certificate. It does not
+// contain the certificate's CN, because these records are indexed by CN.
+message BlacklistedCertificate {
 	// Expiry is the latest known expiration time of a certificate that
-	// was issued to this node.
-	Timestamp expiry = 2;
+	// was issued for the given CN.
```

Comment:
You probably want the Certificate Serial number in here, if you're taking this "blocking Certificates instead of nodes" thing seriously.